src/Npgsql/NpgsqlCommand.cs

```diff
@@ -824,88 +857,84 @@ void CleanupSend()
                 SynchronizationContext.SetSynchronizationContext(null);
         }
 
-        async Task SendExecute(NpgsqlConnector connector, bool async)
+        internal Task Write(NpgsqlConnector connector, bool async)
         {
-            BeginSend();
+            return (_behavior & CommandBehavior.SchemaOnly) == 0
+                ? WriteExecute(connector, async)
+                : WriteExecuteSchemaOnly(connector, async);
 
-            for (var i = 0; i < _statements.Count; i++)
+            async Task WriteExecute(NpgsqlConnector connector, bool async)
```

Comment:
Can you be more specific? Do you mean to inline this into MultiplexingWriteLoop? It's certainly possible, though it would be duplicated with the non-multiplexing path. If this is what you're proposing, I'll look at this more specifically next time I profile.

Reply:
Curious what you find!

Reply:
FYI going to defer this for now... We're already at around 25% improvement after everything, and I don't want to duplicate code prematurely etc... But definitely open to exploring further refactoring after this PR.